### PR TITLE
Dltp 686 add routing option for json ld 

### PR DESCRIPTION
**Pull request by laritakr**
## Claiming issue DLTP-686

@e1257c78db46097b12ec528fe618e94cf5d41c09

$ ./script/start-issue DLTP-686

[skip ci]

## Register mimetype .jsonld

@0f3280fce5a3feea9fb2611a176a4bd192016c49

## Add rendering for jsonld

@ff9ba66b8d0bbbc39d6b9a22fc00a005936e763b

## Add jsonld format to controller

@9caa6457fbbd956332ec1ddb87f55cd204fd7774

## Specs for json mapper

@a1cf8b71c2e717ad9ed4ebc912a729a073e36b4a

## Correct with_json_mapper shared spec

@2486fa655f829577941e86a3ee3f685a2b9bb789

## Closing issue DLTP-686

@cf26a991b304e1e1ed160afdfd4ca0a84c0144c3

$ ./script/close-issue DLTP-686

### CODE REVIEW

File: spec/mapper/datastream_json_mapper_spec.rb [hunk 1]
```diff
@@ -7,50 +7,51 @@ def strip_namespace(pid)
...
-        "nd:accessEdit"=>["#{user.username}"],
-        "dc:abstract"=>"#{curation_concern.abstract}",
-        "dc:created"=>"#{curation_concern.date_modified.strftime("%F")}",
-        "dc:creator#author"=>"#{curation_concern.creator.to_sentence}",
-        "dc:dateSubmitted"=> {"@value"=>"#{curation_concern.date_uploaded.strftime("%FZ")}", "@type"=>"http://www.w3.org/2001/XMLSchema#date"},
-        "dc:language"=>"#{curation_concern.language.to_sentence}",
-        "dc:modified"=> {"@value"=>"#{curation_concern.date_modified.strftime("%FZ")}", "@type"=>"http://www.w3.org/2001/XMLSchema#date"},
-        "dc:rights"=>"#{curation_concern.rights}",
-        "dc:title"=>"#{curation_concern.title}"}
+    {"@context"=> {"und"=>"#{Rails.configuration.application_root_url}/show/",
```
Comment (jeremyf) [inline spec/mapper/datastream_json_mapper_spec.rb]: Replace with the following:

```ruby
{"@context"=> {"und"=> File.join(Rails.configuration.application_root_url}, "show/"),
```

This way we don't have to know whether `Rails.configuration.application_root_url` resolves to a string that ends in `/` or not.

File: spec/mapper/datastream_json_mapper_spec.rb [hunk 2]
```diff
@@ -7,50 +7,51 @@ def strip_namespace(pid)
...
+     "dc:language"=>"#{curation_concern.language.to_sentence}",
+     "dc:modified"=> {"@value"=>"#{curation_concern.date_modified.strftime("%FZ")}", "@type"=>"http://www.w3.org/2001/XMLSchema#date"},
+     "dc:rights"=>"#{curation_concern.rights}",
+     "dc:title"=>"#{curation_concern.title}"
+    }
   }
 
   let(:expected_generic_file_json){
-    {"@context"=> {"bibo"=>"http://purl.org/ontology/bibo/",
+    {"@context"=> {"und"=>"#{Rails.configuration.application_root_url}/show/",
```
Comment (jeremyf) [inline spec/mapper/datastream_json_mapper_spec.rb]: Replace with the following:

```ruby
{"@context"=> {"und"=> File.join(Rails.configuration.application_root_url}, "show/"),
```

This way we don't have to know whether `Rails.configuration.application_root_url` resolves to a string that ends in `/` or not.

File: spec/mapper/datastream_json_mapper_spec.rb [hunk 3]
```diff
@@ -72,16 +73,16 @@ def strip_namespace(pid)
                    "previousVersion"=>"http://purl.org/pav/previousVersion"},
      "@id"=>"#{generic_file.id}",
      "nd:afmodel"=>"#{generic_file.class.name}",
-     "dc:dateSubmitted"=> {"@value"=>"#{generic_file.date_uploaded.strftime("%FZ") }", "@type"=>"http://www.w3.org/2001/XMLSchema#date"},
+     "dc:dateSubmitted"=> {"@value"=>"#{generic_file.date_uploaded.strftime("%FZ")}", "@type"=>"http://www.w3.org/2001/XMLSchema#date"},
      "dc:modified"=> {"@value"=>"#{generic_file.date_modified.strftime("%FZ")}", "@type"=>"http://www.w3.org/2001/XMLSchema#date"},
      "isPartOf" => "#{curation_concern.id}",
      "nd:accessEdit" => ["#{user.username}"],
-     "nd:content" => "https://curate.nd.edu/downloads/#{strip_namespace(generic_file.pid)}",
+     "nd:content" => "#{Rails.configuration.application_root_url}/downloads/#{strip_namespace(generic_file.pid)}",
```
Comment (jeremyf) [inline spec/mapper/datastream_json_mapper_spec.rb]: Replace with the following:

```ruby
"nd:content" => File.join(Rails.configuration.application_root_url, "downloads", strip_namespace(generic_file.pid))
```

This way we don't have to know whether `Rails.configuration.application_root_url` resolves to a string that ends in `/` or not.

File: spec/mapper/datastream_json_mapper_spec.rb [hunk 4]
```diff
@@ -72,16 +73,16 @@ def strip_namespace(pid)
...
      "nd:depositor" => "#{user.username}",
      "nd:filename" => "#{generic_file.filename}",
      "nd:mimetype" => "#{generic_file.content.mimeType}",
      "nd:owner" => "#{user.username}",
-     "nd:thumbnail" => {"@id"=>"https://curate.nd.edu/downloads/#{strip_namespace(generic_file.pid)}/thumbnail"}
+     "nd:thumbnail" => {"@id"=>"#{Rails.configuration.application_root_url}/downloads/#{strip_namespace(generic_file.pid)}/thumbnail"}
```
Comment (jeremyf) [inline spec/mapper/datastream_json_mapper_spec.rb]: Replace with the following:

```ruby
"nd:content" => File.join(Rails.configuration.application_root_url, "downloads", strip_namespace(generic_file.pid), 'thumbnail')
```

This way we don't have to know whether `Rails.configuration.application_root_url` resolves to a string that ends in `/` or not.

Comment (laritakr) [inline spec/mapper/datastream_json_mapper_spec.rb]: Ok changed.

**Pull request merged**